pkg/policy/rule.go

```diff
@@ -124,7 +124,7 @@ func mergeL4Port(ctx *SearchContext, fromEndpoints []api.EndpointSelector, r api
 		}
 	}
 
-	if adjustL4PolicyIfNeeded(fromEndpoints, &v) && len(r.Rules.HTTP) == 0 {
+	if adjustL4PolicyIfNeeded(fromEndpoints, &v) && r.Rules != nil && r.Rules.Len() == 0 {
```

Comment:
Let's do `r.Rules.NumRules()` and you can check the pointer passed in against `nil`

Reply:
OK.